src/cli.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{CliArgs, Client, GenesisDetails, StateId};
+use ethereum_consensus::{phase0::mainnet::Fork, primitives::Root};
+use url::Url;
+
+pub async fn cli(args: CliArgs) {
+    //set up client
+    let s = args.node_url;
+    let url: Url = Url::parse(&s).unwrap();
+    let client = Client::new(url);
+    let endpoint = args.endpoint.as_str();
+
+    match endpoint {
+        // Match on the provided endpoint
+        "node_version" => println!("{:?}", client.get_node_version().await.unwrap()),
+        "genesis" => {
```

Comment:
it would be nice to have stronger-typing here by using `clap` subcommands with bespoke `enum` types to map out all of the possible data

see the `mev-rs` code I linked below for an example there